hotel/models.py

```diff
@@ -0,0 +1,137 @@
+    hotel_requests = relationship('HotelRequests', backref=backref('attendee', load_on_pending=True), uselist=False)
+    room_assignments  = relationship('RoomAssignment', backref=backref('attendee', load_on_pending=True), uselist=False)
+
+    @presave_adjustment
+    def staffer_hotel_eligibility(self):
+        if self.badge_type == c.STAFF_BADGE:
+            self.hotel_eligible = True
+
+    @presave_adjustment
+    def staffer_setup_teardown(self):
+        if self.setup_hotel_approved:
+            self.can_work_setup = True
+        if self.teardown_hotel_approved:
+            self.can_work_teardown = True
+
+    @property
+    def hotel_shifts_required(self):
+        return bool(c.SHIFTS_CREATED and self.hotel_nights and self.ribbon != c.DEPT_HEAD_RIBBON and self.takes_shifts)
+
+    @property
+    def setup_hotel_approved(self):
+        hr = self.hotel_requests
+        return bool(hr and hr.approved and set(hr.nights_ints).intersection(c.SETUP_NIGHTS))
+
+    @property
+    def teardown_hotel_approved(self):
+        hr = self.hotel_requests
+        return bool(hr and hr.approved and set(hr.nights_ints).intersection(c.TEARDOWN_NIGHTS))
+
+    @property
+    def shift_prereqs_complete(self):
+        return not self.placeholder and self.food_restrictions and self.shirt_size_marked \
+            and (not self.hotel_eligible or self.hotel_requests or not c.BEFORE_ROOM_DEADLINE)
+
+    @property
+    def hotel_nights(self):
+        try:
+            return self.hotel_requests.nights
+        except:
+            return []
+
+    @cached_property
+    def hotel_status(self):
+        hr = self.hotel_requests
+        if not hr:
+            return 'Has not filled out volunteer checklist'
+        elif not hr.nights:
+            return 'Declined hotel space'
+        elif hr.setup_teardown:
+            return 'Hotel nights: {} ({})'.format(hr.nights_display, 'approved' if hr.approved else 'not yet approved')
+        else:
+            return 'Hotel nights: ' + hr.nights_display
+
+
+class HotelRequests(MagModel, NightsMixin):
+    attendee_id        = Column(UUID, ForeignKey('attendee.id'), unique=True)
+    nights             = Column(MultiChoice(c.NIGHT_OPTS))
+    wanted_roommates   = Column(UnicodeText)
+    unwanted_roommates = Column(UnicodeText)
+    special_needs      = Column(UnicodeText)
+    approved           = Column(Boolean, default=False, admin_only=True)
+
+    def decline(self):
+        self.nights = ','.join(night for night in self.nights.split(',') if int(night) in c.CORE_NIGHTS)
+
+    def __repr__(self):
+        return '<{self.attendee.full_name} Hotel Requests>'.format(self=self)
+
+
+class Room(MagModel, NightsMixin):
+    notes      = Column(UnicodeText)
+    locked_in  = Column(Boolean, default=False)
+    nights     = Column(MultiChoice(c.NIGHT_OPTS))
+    created    = Column(UTCDateTime, server_default=utcnow())
+    assignments = relationship('RoomAssignment', backref='room')
+
```

Comment:
Since our ``AutomatedEmail`` code works on anything with an ``email`` field, we just define one here to include everyone in the room.  That worked perfectly with no modifications needed to the core repo :)